src/shared/directives/conditional-show/domain/conditional-show.model.ts

```diff
@@ -56,11 +60,34 @@ export class ShowCondition {
     }
   }
 
-  private findValueForComplexCondition(fields: any, head: string, tail: string[]) {
+  private findValueForComplexCondition(fields: any, head: string, tail: string[], path?: string) {
     if (tail.length === 0) {
       return fields[head];
     } else {
-      return this.findValueForComplexCondition(fields[head], tail[0], tail.slice(1));
+      if (Array.isArray(fields[head])) {
+        // use the path to resolve which array element we refer to
+        if (path.startsWith(head)) {
```

Comment:
will it be better to create a separate function for this if block (if logical to do so)?

Reply:
I will leave it this way, because IDE doesn't want to do it for me :)